luigi/execution_summary.py

```diff
@@ -410,41 +467,34 @@ def _summary_format(set_tasks, worker):
         if len(ext_workers) == 0:
             str_output += '\n'
         str_output += 'Did not run any tasks'
-    smiley, reason = _tasks_status(set_tasks)
-    str_output += "\nThis progress looks {0} because {1}".format(smiley, reason)
+    summary_text_one_line, _ = _tasks_status(set_tasks)
+    str_output += "\n{0}".format(summary_text_one_line)
     if num_all_tasks == 0:
         str_output = 'Did not schedule any tasks'
     return str_output
 
 
 def _tasks_status(set_tasks):
     """
-    Given a grouped set of tasks, returns a smiley and a readable reason for the expression
+    Given a grouped set of tasks, returns a one line summary and a LuigiRetCode
     """
-    smiley = ""
-    reason = ""
     if set_tasks["ever_failed"]:
         if not set_tasks["failed"]:
-            smiley = ":)"
-            reason = "there were failed tasks but they all succeeded in a retry"
+            retcode = LuigiRetCodes.SUCCESS_WITH_RETRY
         else:
-            smiley = ":("
-            reason = "there were failed tasks"
+            retcode = LuigiRetCodes.FAILED
             if set_tasks["scheduling_error"]:
-                reason += " and tasks whose scheduling failed"
+                retcode = LuigiRetCodes.FAILED_AND_SCHEDULING_FAILED
     elif set_tasks["scheduling_error"]:
-        smiley = ":("
-        reason = "there were tasks whose scheduling failed"
+        retcode = LuigiRetCodes.SCHEDULING_FAILED
     elif set_tasks["not_run"]:
-        smiley = ":|"
-        reason = "there were tasks that were not granted run permission by the scheduler"
+        retcode = LuigiRetCodes.NOT_RUN
     elif set_tasks["still_pending_ext"]:
-        smiley = ":|"
-        reason = "there were missing external dependencies"
+        retcode = LuigiRetCodes.MISSING_EXT
```

Comment:
Maybe even nicer with a helper method so all these `retcode = XXX` can be converted into `return XXX`, so you don't make any control flow bugs.